src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocket.cs

```diff
@@ -113,6 +113,7 @@ public override string SubProtocol
                         secureConnection ? Interop.WinHttp.WINHTTP_FLAG_SECURE : 0);
 
                     ThrowOnInvalidHandle(_operation.RequestHandle);
+                    _operation.IncrementHandlesOpenWithCallback();
```

Comment:
I don't know the details of the callbacks here, but is there any chance that between the time you call Increment here and the time you call Pin below, that something could go wrong, cause Pin to not be called, but have the callback that does a decrement and an unpin still be called?  I'm assuming not, just verifying.